internal/pkg/templating/controllerWatchFunctionTemplate.go

```diff
@@ -0,0 +1,59 @@
+package templating
+
+import (
+	"bytes"
+	"text/template"
+
+	"github.com/redhat-nfvpe/helm2go-operator-sdk/internal/resourcecache"
+)
+
+// ControllerWatchFuncConfig contains the necessary fields to render the controller watch function specific to a resource
+type ControllerWatchFuncConfig struct {
+	APIVersion            string
+	ResourceImportPackage string
+	ResourceType          string
+	Kind                  string
+	LowerKind             string
+	OwnerAPIVersion       string
+}
+
+// NewControllerWatchFuncConfig returns a new controller watch function configuration object
+func NewControllerWatchFuncConfig(apiVersion, ownerAPIVersion, kind, lowerKind string, r *resourcecache.Resource) *ControllerWatchFuncConfig {
```

Comment:
Again, certain responsibilities within the `template.go` function should be encapsulated in this function.